spec/lib/contextual_logger_spec.rb

```diff
@@ -13,26 +13,88 @@ def expect_log_line_to_be_written(log_line)
   match { |actual| JSON.parse(actual) == JSON.parse(expected) }
 end
 
-describe 'ContextualLogging' do
+describe ContextualLogger do
   before(:each) { Time.now_override = Time.now }
   after(:each)  { logger.global_context = {} }
 
   subject(:logger) { ContextualLogger.new(Logger.new('/dev/null')) }
 
   it { is_expected.to respond_to(:with_context) }
-  it { is_expected.to respond_to(:debug) }
-  it { is_expected.to respond_to(:info) }
-  it { is_expected.to respond_to(:warn) }
-  it { is_expected.to respond_to(:error) }
-  it { is_expected.to respond_to(:fatal) }
+
+  context 'log_level' do
+    let(:log_stream) { StringIO.new }
+    let(:default_logger) { ContextualLogger.new(Logger.new(log_stream)) }
+    let(:log_level) { Logger::Severity::DEBUG }
+    let(:logger) { ContextualLogger.new(Logger.new(log_stream, level: log_level)) }
+
+    context 'at default level' do
+      it 'respects log level debug' do
+        log_at_every_level(default_logger)
+        expect(log_message_levels).to eq(["debug", "info", "warn", "error", "fatal", "unknown"])
+      end
+    end
+
+    context 'at level debug' do
+      let(:log_level) { Logger::Severity::DEBUG }
+
+      it 'respects log level' do
+        log_at_every_level(logger)
+        expect(log_message_levels).to eq(["debug", "info", "warn", "error", "fatal", "unknown"])
+      end
+    end
+
+    context 'at level info' do
+      let(:log_level) { Logger::Severity::INFO }
+
+      it 'respects log level' do
+        log_at_every_level(logger)
+        expect(log_message_levels).to eq(["info", "warn", "error", "fatal", "unknown"])
+      end
+    end
+
+    context 'at level warn' do
+      let(:log_level) { Logger::Severity::WARN }
+
+      it 'respects log level' do
+        log_at_every_level(logger)
+        expect(log_message_levels).to eq(["warn", "error", "fatal", "unknown"])
+      end
+    end
+
+    context 'at level error' do
+      let(:log_level) { Logger::Severity::ERROR }
+
+      it 'respects log level' do
+        log_at_every_level(logger)
+        expect(log_message_levels).to eq(["error", "fatal", "unknown"])
+      end
+    end
+
+    context 'at level fatal' do
+      let(:log_level) { Logger::Severity::FATAL }
+
+      it 'respects log level' do
+        log_at_every_level(logger)
+        expect(log_message_levels).to eq(["fatal", "unknown"])
+      end
+    end
+
+    context 'at level unknown' do
+      let(:log_level) { Logger::Severity::UNKNOWN }
+
+      it 'respects log level' do
+        log_at_every_level(logger)
+        expect(log_message_levels).to eq(["unknown"])
+      end
+    end
+  end
```

Comment:
I was surprised that the above tests weren't there before!

Reply:
Because before log level wasn't ever being modified by the gem, so log level should act however it acts in the logger being extended. You've just now introduced knowledge of log level to this gem, making this tests a requirement.

Reply:
True.